ceph_deploy/util/init_systems.py

```diff
@@ -0,0 +1,246 @@
+from ceph_deploy.lib import remoto
+
+# Provide mappings/translations for services depending
+# on Init system.  All services coming into the classes
+# here should be one of ceph-mon, ceph-osd, ceph-mds, or
+# ceph-radosgw. It's then up to the code here to Do The
+# Right Thing.
+SYSV_SERVICES = {
+    'ceph-mon': 'mon',
+    'ceph-mds': 'mds'
+}
+
+UPSTART_SERVICES = {
+    'ceph-radosgw': 'radosgw'
+}
+
+
+class InitSystem(object):
+    """
+    Base class for all Init systems
+    """
+
+    def __init__(self, remote_conn):
+        self.remote_info = remote_conn
+        self.remote_conn = remote_conn.conn
+        self.hostname = self.remote_conn.remote_module.shortname()
+
+    def _run(self, cmd, **kw):
+        if 'timeout' not in kw:
+            kw['timeout'] = 7
+        return remoto.process.run(
+            self.remote_conn,
+            cmd,
+            **kw
+        )
+
+    def start(self, service, **kw):
+        """Start service on remote node"""
+        raise NotImplementedError()
+
+    def stop(self, service, **kw):
+        """Stop service on remote node"""
+        raise NotImplementedError()
+
+    def enable(self, service, **kw):
+        """Enable service on remote node"""
+        raise NotImplementedError()
+
+    def disable(self, service, **kw):
+        """Disable service on remote node"""
+        raise NotImplementedError()
+
+    def status(self, service, **kw):
+        """Get service status on remote node"""
+        raise NotImplementedError()
+
+
+class SysV(InitSystem):
+    """
+    SysV Init System
+    """
+
+    name = 'sysvinit'
+
+    def __init__(self, remote_conn):
+        super(SysV, self).__init__(remote_conn)
+        self.service_exe = self.remote_conn.remote_module.which_service()
+
+    def _get_init_script(self, service):
+        return 'ceph-radosgw' if service == 'ceph-radosgw' else 'ceph'
+
+    def _build_service_command(
```

Comment:
@alfredodeza Thanks for pointing out the eager behavior in `__init__`.  I've removed that in the two places I had it and used `@property` instead.  A definite improvement.